<Feature: Villager Path Finding>

projojoboy opened the pull request.

Commit:
<Feature: Villager Path Finding>
Add Idle State
Add Idle Actions
Add Wander State
Add Wander Actions
Add Timer Decisions
Add Destination Reached Decisions
Add WalkTowards State
Add Test Scene
Add Material Red
Add Material Blue
Add Material Green
Add Material Gray

Commit:
Merge branch 'development' into feature/villager-path-finding

# Conflicts:
#	Assets/Scripts/Features.meta

empireboy commented on Assets/Scripts/Features/AI/Villager Actions/WanderingAction.cs:
```diff
@@ -0,0 +1,64 @@
+        NavMesh.SamplePosition(randomDirection, out hit, 50, NavMesh.AllAreas);
+
+        Vector3 destination = hit.position;
+        return destination;
```
Here you can just do return hit.position instead of saving the position inside a new Vector3 before returning.

empireboy reviewed (changes requested):
There are some issues that I came across. These issues are mostly related to folder structure.
Let me point them out:

**- The location for files and folders inside the folder Assets/AI/Villager needs to be changed.**
Here is an example that will make more sense:

> Assets
>    >    AI
>    >    >    States
>    >    >    Decisions
>    >    >    Actions
>    >    >    Villager
>    >    >    >    States
>    >    >    >    Decisions
>    >    >    >    Actions

Note that the states/decisions/actions folders inside the AI folder are global states/decisions/actions.
The ones inside the Villager folder are only related to the Villager.

Also note that the actions inside Scripts/Actions can be placed into their corresponding actions folder, because these actions are also ScriptableObjects and not scripts.

**- The location for your testing scene needs to be changed.**
Here is an example for the scene structure:

> Assets
>    >    Scenes
>    >    >    Development Scenes
>    >    >    >    Villager AI Testing
>    >    >    >    >    VillagerAITestingScene (this is the scene file)
>    >    >    >    >    NavMesh (this is the NavMesh file)

**- You missed some dots at the end of your comments.**
I can add the dots aswell if you make sure that everything else is fixed.

**- For the scripts folder you can have the same folder structure as shown above for the AI folder.**
So if you need villager decision scripts you could find them in Scripts/Features/AI/Villager/Decisions

**- You can remove the Ground layer.**

**- If everything else is fixed you can create a prefab from the Villager object and place it inside Assets/Prefabs/Villagers/SomeVillagerPrefabName. You can add this prefab to the GameScene, and add a plane that it can walk over.**

Commit:
<Feature: Villager Path Finding>
Update pull request, changes request

Commit:
<Feature: Villager Path Finding>
Add AI to main scene
Add Little tweaks to make the AI movement veel better
Add New map structure

empireboy committed:
<Feature: Villager Path Finding>
Add Gizmo for targetLocation

empireboy reviewed (changes requested):
The folder structure is fixed now, a bug that I came accross is that all villagers stopped moving when in the Wandering state. I also added a gizmo for the targetLocation.

Commit:
Merge branch 'development' into feature/villager-path-finding

Commit:
<Feature: Villager Path Finding>
Undo scene changes
